### ISSUE TITLE
Unable to sync Group Sharepoint documents

### ISSUE THREAD

**Issue by arm1e** [abraunegg/onedrive#668]
### Bug Report Details ###
Trying to sync with a sharepoint site  that I own on onedrive for business but get 'access denied' and 'forbidden' when attempting to get the drive id. 

**Application and Operating System Details:**
*   OS: 
````bash
Linux 5.0.0-29-generic #31-Ubuntu SMP Thu Sep 12 13:05:32 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux (kubuntu 19.04)
````

*   OneDrive Account Type: Business (school)

*   Application configuration: 
````bash
onedrive --display-config
onedrive version                    = v2.2.5-32-g0d3b891
Config path                         = /home/steven/.config/onedrive
Config file found in config path    = false
Config option 'sync_dir'            = /home/steven/OneDrive
Config option 'skip_file'           = ~*
Config option 'skip_symlinks'       = false
Config option 'monitor_interval'    = 45
Config option 'min_notif_changes'   = 5
Config option 'log_dir'             = /var/log/onedrive/
Selective sync configured           = false
````

*   Curl Version: Output of `curl --version`
````bash
curl 7.64.0 (x86_64-pc-linux-gnu) libcurl/7.64.0 OpenSSL/1.1.1b zlib/1.2.11 libidn2/2.0.5 libpsl/0.20.2 (+libidn2/2.0.5) libssh/0.8.6/openssl/zlib nghttp2/1.36.0 librtmp/2.3
Release-Date: 2019-02-06
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL 
````
**Complete Verbose Log Output**

```bash
Loading config ...
Using Config Dir: /home/steven/.config/onedrive
No config file found, using application defaults
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/steven/OneDrive
Initializing the Synchronization Engine ...
Account Type: business
Default Drive ID: b!bICyIFS2gUqjwGbxFHNkrqa8mqBUjBRLmPp11aRGbAuIhJrH6vXtQ4eB98UX_4NM
Default Root ID: 01WRBLKNN6Y2GOVW7725BZO354PWSELRRZ
Remaining Free Space: 1083133389977
Fetching details for OneDrive Root
OneDrive Root exists in the database
onedrive.OneDriveException@src/onedrive.d(713): HTTP request returned status code 403 (Forbidden)
{
    "error": {
        "code": "accessDenied",
        "innerError": {
            "date": "2019-09-21T19:52:40",
            "request-id": "91d10c96-116d-45f5-9840-1e65751f8045"
        },
        "message": "The caller does not have permission to perform the action."
    }
}
----------------
src/onedrive.d:48 void onedrive.OneDriveApi.checkHttpCode(ref const(std.json.JSONValue)) [0x10f95d65]
src/onedrive.d:393 std.json.JSONValue onedrive.OneDriveApi.get(const(char)[], bool) [0x10f94eab]
src/onedrive.d:327 std.json.JSONValue onedrive.OneDriveApi.o365SiteSearch(immutable(char)[]) [0x10f94997]
src/sync.d:1946 void sync.SyncEngine.querySiteCollectionForDriveID(immutable(char)[]) [0x10fa77aa]
src/main.d:449 _Dmain [0x10f7d695]

```

**Comment by arm1e** [comment]
I can confirm that I am set at the owner of this site and it also syncs on my work windows machine via the onedrive application

**Comment by abraunegg** [comment]
You are running v2.2.5 from Ubuntu packages. This is very old and unsupported.

Pleae upgrade your client to v2.3.9 or above using the instructions as per install.md

**Comment by arm1e** [comment]
Thank you so much! I did not realise that I had forgotten to reinstall when I wiped my machine, I must have just reinstalled my old git clone. Doh!

**Comment by abraunegg** [comment]
@arm1e 
I guess this is a pitfall of using a 'clone' as this becomes a very static point in time copy of the origin.

**Comment by abraunegg** [comment]
Closing issue, yes a bug (#619) , but fixed in v2.3.9 (#620)
